content/en/blog/2023-06-23-unikraft-gsoc-enhancing-vscode-developer-experience.md

```diff
@@ -0,0 +1,48 @@
++++
+title = "GSoC'23: Enhancing VScode Developer Experience"
+date = "2023-06-23T00:00:00+01:00"
+author = "Md Sahil"
+tags = ["GSoC'23", "VsCode", "IDE Extension"]
++++
+
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-2022-badge.svg" align="right" />
+
+# GSoC'23: Enhancing VScode Developer Experience
+
+## Motivation
+
+The VsCode Extension for unikraft enables developers to quickly and painlessly build unikernels from the VsCode IDE.
+Amongst other features, It allows developers to list and find Unikernel libraries as well as run basic commands.
+
+In this project, I will upgrade the VSCode extension to use KraftKit, the newly released CLI
+companion tool for unikraft rewritten in GO.
+
+### Summary of Objectives
+
+This project aims to achieve the following outcomes:
+
+- The project requires modifications to the project’s main binary, kraft, To enable JSON output of various commands so that the integration with
+VsCode could be done through a machine-readable interface.
+
+- And Updating VsCode IDE extension to function properly with new golang based Kraftkit.
+
+- Additionally, I will enhance the experience, including adding support for additional steps in Unikraft’s build cycle; Packaging unikernels in different formats; Providing a linting.
+
+## Current Progress
+
+- Updated kraftkit to [output in configurable format](https://github.com/unikraft/kraftkit/pull/499) (JSON | YAML) for the commands `kraft pkg list` and `kraft ps`.
+
+- [Updated VsCode IDE Extension's Unikraft-view](https://github.com/unikraft/ide-vscode/pull/7) to list available local components in the system and in the workspace's Kraftfile.
+
+## Next steps
+
+- To create a subcommand `show` for command `kraft pkg` in kraftkit to produce information about a component like happens in pykraft by the command `kraft list show`.
+
+- To make listing of versions possible via IDE Extension when to add a new library/component using IDE Exntension.
+
+- To enable IDE to execute Initialization operation & others to work properly with kraftkit.
+
+
+## About Me
+
+I'm Md Sahil, A Third year Undergraduate Student of Computer Science Engineering as Maharshi Dayanand University.
```

Comment:
```suggestion
I'm Md Sahil, A third year undergraduate student of Computer Science Engineering at Maharshi Dayanand University.
```

Reply:
Link your name to your GitHub URL: https://github.com/MdSahil-oss

Also add a link to the university